airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -216,14 +216,19 @@ def _read_stream(
             stream_instance.state = stream_state  # type: ignore # we check that state in the dir(stream_instance)
             logger.info(f"Setting state of {self.name} stream to {stream_state}")
 
-        record_iterator = stream_instance.read(
-            configured_stream,
-            logger,
-            self._slice_logger,
-            stream_state,
-            state_manager,
-            internal_config,
-        )
+        # Successful resumable full refresh streams emit a sentinel value to indicate that it should be skipped on subsequent
+        # attempts. Full refresh stream state is cleared in between jobs so this is never present on the first attempt
+        if stream_state == {"__ab_full_refresh_sync_complete": True}:
```

Comment:
I don't love having such specific RFR logic leak into the `abstract_source.py` classes.

The alternative is to push it a little lower into potentially the `CheckpointReader`s, but it does require covering more codepaths

Reply:
how much more complexity would it add to push it down? I don't feel the need to be dogmatic about it, but  can we at least make this dict a shared constant?

Reply:
I actually did end up shifting this pattern. I like the design better in the checkpoint reader since its the natural place to interpret state and decide whether to sync.

The other nice benefit it has is it moves us into a more consistent paradigm for the platform. Every stream is still synced (instead of truly skipped) so it has stream status and emits at least one state message w/ count 0. I've updated things accordingly